File: Leetcode/27_remove_element/solution.go
package removeelement

func removeElement(nums []int, val int) int {
	postion := 0

	for idx, num := range nums {
		if num != val {
			nums[postion] = nums[idx]
			postion++
		}
	}

	return postion
}

func main() {
	nums := []int{0, 1, 2, 2, 3, 0, 4, 2}
	removeElement(nums, 2)
}

File: Leetcode/27_remove_element/solution.js
function removeElement(nums, val) {
  let position = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== val) {
      nums[position] = nums[i];
      position++;
    }
  }
  return position;
}

console.log(removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2));
